### PR TITLE
Add CyclicPattern class for generation of payload data

### PR DESCRIPTION
**Pull request by polybassa**
I think its useful to have such a cyclic pattern.

### CODE REVIEW

**Comment by polybassa** [comment]
import.uts failed. Reopen/close trick.

**Comment by polybassa** [comment]
Ready for review

**Comment by polybassa** [comment]
unrelated error, apply reopen close

**Comment by polybassa** [comment]
reopen close trick.. an ISOTPSoftSocket test failed. I've tried this testcase more than 2000 times on my local system but couldn't reproduce an error. 
```
Regression tests for ISOTPSoftSocket
-- Run at 18:55:19 from [test/contrib/isotp_soft_socket.uts] by UTscapy in 19.7728979588
 -> Passed=51
 -> Failed=1

######
## More complex operations
######


###(037)=[failed] ISOTPSoftSocket sniff

>>> msg = ISOTP(b'\x11\x22\x33\x11\x22\x33\x11\x22\x33\x11\x22\x33')
>>> with TestSocket(CAN) as isocan1, ISOTPSoftSocket(isocan1, 0x123, 0x321) as sock, \
...         TestSocket(CAN) as isocan, ISOTPSoftSocket(isocan, 0x321, 0x123) as rx_sock:
...     isocan1.pair(isocan)
...     msg.data += b'0'
...     sock.send(msg)
...     msg.data += b'1'
...     sock.send(msg)
...     msg.data += b'2'
...     sock.send(msg)
...     msg.data += b'3'
...     sock.send(msg)
...     msg.data += b'4'
...     sock.send(msg)
...     rx = rx_sock.sniff(count=5, timeout=5)
... 
TX queue not empty
>>> msg = ISOTP(b'\x11\x22\x33\x11\x22\x33\x11\x22\x33\x11\x22\x33')
>>> msg.data += b'0'
>>> assert(rx[0] == msg)
Traceback (most recent call last):
  File "<input>", line 2, in <module>
  File "scapy/plist.py", line 176, in __getitem__
    return self.res.__getitem__(item)
IndexError: list index out of range
```

**Comment by polybassa** [comment]
@gpotter2 Could you please review and merge this PR?

File: scapy/volatile.py [hunk 1]
```diff
@@ -1409,3 +1413,121 @@ class CorruptedBits(CorruptedBytes):
...
+        if charset_type == 1:  # extended type
+            charset[1] = "%$-;" + re.sub("[sn]", "", charset[1])
+            charset[2] = "sn()" + charset[2]
+
+        if charset_type == 2:  # maximum type
+            charset += [string.punctuation]
+
+        mixed_charset = ""
+        if six.PY2:
+            for k in itertools.izip_longest(*charset, fillvalue=""):
```
Comment (gpotter2) [inline scapy/volatile.py:1461]: Pretty sure there's a six for that

**Comment by polybassa** [comment]
I could not found a similar function in six
-- 
Viele Grüße

Nils Weiß

On February 9, 2022 5:51:48 PM GMT+01:00, gpotter2 ***@***.***> wrote:
***@***.*** commented on this pull request.

File: scapy/volatile.py (hunk 1 above)
Comment (polybassa) [reply scapy/volatile.py]: Found it!

**Review by gpotter2 — APPROVED** [review]

**Comment by polybassa** [comment]
@gpotter2 could you please merge this and maybe some other PRs?

On February 13, 2022 9:33:32 PM GMT+01:00, gpotter2 ***@***.***> wrote:
***@***.*** approved this pull request.

**Review by KhazAkar — APPROVED** [review]
LGTM

**Pull request merged**